src/lib/meta/meta_blk_service.cpp

```diff
@@ -375,7 +377,12 @@ void MetaBlkService::register_handler(meta_sub_type type, const meta_blk_found_c
     m_sub_info[type].cb = cb;
     m_sub_info[type].comp_cb = comp_cb;
     m_sub_info[type].do_crc = do_crc ? 1 : 0;
-    HS_LOG(INFO, metablk, "[type={}] registered with do_crc: {}", type, do_crc);
+    m_dep_topo_graph[type];
```

Comment:
this is not needed. 
if (deps) {
    m_sub_info[type].no_dep = false;
    for (auto const& x : deps) {
        m_sub_info[x].no_deps = false;
    }
}

Reply:
@yamingk that looks good through seems we dont need to maintain the `no_dep`
```
std::unordered_set<std::string> m_independent_subtypes;
```
```
if (deps.has_value()) {
   for (auto const& x : deps.value()) {
        m_independent_subtypes.erase(x);
        m_dep_topo_graph[x].push_back(type);
    }
    m_dep_topo_graph[type];
 } else {
     m_independent_subtypes.insert(type);
```

Reply:
no. when client register , it only know whether it depends on others, it does not know whether others depend on it until all the register action finishes. what`s more, the way we determine whether a subtype is independent is to put all the subtype in the graph and check the in_degree and out_degree. only those subtypes with both 0 in_degree and 0 out_degree are actually independent

Reply:
One can argue either way, right, why do we need another set compared to using existing one?